src/components/ui/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { LinearProgress, LinearProgressProps } from "react-native-elements";
+import { Text, View } from "react-native";
+import { useStyles } from "./styles";
+import { colors } from "@app/utils";
+
+interface ProgressBarProps
+  extends Omit<LinearProgressProps, "value" | "color" | "variant"> {
+  id: string;
+  type: "info" | "warning" | "success";
+  maxValue: number;
+  curValue?: number;
+  label?: string;
+  hasProgressText?: boolean; // show progress in label
+  labelType?: "info" | "warning" | "success" | "default";
```

Comment:
should these enums be a generic type?

Reply:
I was thinking of putting it in a generic type too but not every component would have all options I think